app/change_set_persisters/plum_change_set_persister.rb

```diff
@@ -52,6 +52,12 @@ def with(metadata_adapter:)
     def before_save(change_set:)
       apply_remote_metadata(change_set: change_set)
       create_files(change_set: change_set)
+
+      members(change_set: change_set).each do |member|
+        member.read_groups = change_set.read_groups
```

Comment:
We should probably only do this if the read_groups or state have changed, otherwise, we'll be touching all the FileSets in a ScannedResource any time we make a metadata change.